### PR TITLE
Fix remaining bugs

### PR DESCRIPTION
**Pull request by SUPERCILEX**
- Removes unnecessary Bevy features
- Fixes a bug where the FPS meter is updated on every NOT tick instead of every tick (aka 59 times per second)
- Runs cargo fmt and simplifies some stuff

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -88,26 +87,23 @@ fn update(
...
-                    if let Some(fps) = fps_diags {
-                        write!(value, "{}{:.0}", STRING_FORMAT, fps).unwrap();
-                    } else {
-                        value.clear();
-                        write!(value, "{}", STRING_MISSING).unwrap();
-                    }
+        if state.timer.paused() {
+            // Timer is paused so remove text
+            for mut text in text_query.iter_mut() {
+                text.sections[0].value.clear();
```
Comment (jomala) [inline src/lib.rs:93]: This clears the text every frame when the timer is paused. My code does this once only when disable is set.

File: src/lib.rs [hunk 2]
```diff
@@ -126,7 +122,7 @@ fn spawn_text(mut commands: Commands, asset_server: Res<AssetServer>) {
         .spawn_bundle(TextBundle {
             text: Text {
                 sections: vec![TextSection {
-                    value: STRING_INITIAL.to_string(),
+                    value: String::new(),
```
Comment (jomala) [inline src/lib.rs:126]: If someone wanted to update the FPS every 10s, better there is some initial text, I think. Is this an issue for you?

File: src/lib.rs [hunk 3]
```diff
@@ -88,26 +87,23 @@ fn update(
...
-                let fps_diags = extract_fps(&diagnostics);
-
-                for mut text in text_query.iter_mut() {
-                    let value = &mut text.sections[0].value;
-                    value.clear();
-
-                    if let Some(fps) = fps_diags {
-                        write!(value, "{}{:.0}", STRING_FORMAT, fps).unwrap();
-                    } else {
-                        value.clear();
```
Comment (jomala) [inline src/lib.rs:108]: This line of mine could go, true.

File: src/lib.rs [hunk 4]
```diff
@@ -88,26 +87,23 @@ fn update(
     mut text_query: Query<&mut Text, With<ScreenDiagsText>>,
 ) {
     if let Some(mut state) = state_resource {
-        if state.update_now || state.timer.tick(time.delta()).just_finished() {
```
Comment (jomala) [inline src/lib.rs:91]: I seem to have lost and "update_now = false" from here.

File: src/lib.rs [hunk 5]
```diff
@@ -88,26 +87,23 @@ fn update(
...
+
+            for mut text in text_query.iter_mut() {
+                let value = &mut text.sections[0].value;
+                value.clear();
+
+                if let Some(fps) = fps_diags {
+                    state.update_now = false;
+                    write!(value, "{}{:.0}", STRING_FORMAT, fps).unwrap();
+                } else {
+                    write!(value, "{}", STRING_MISSING).unwrap();
```
Comment (jomala) [inline src/lib.rs:107]: Are you keen to get the first real FPS value up ASAP then? Even though it's probably not very accurate yet?

File: Cargo.toml [hunk 6]
```diff
@@ -12,4 +12,7 @@ repository = "https://github.com/jomala/bevy_screen_diags/"
 edition = "2021"
 
 [dependencies]
-bevy = { version = "0.7.0", default-features = false, features = ["render", "bevy_winit", "x11"] }
+bevy = { version = "0.7.0", default-features = false, features = ["bevy_render", "bevy_ui", "bevy_text"] }
+
+[dev-dependencies]
+bevy = { version = "0.7.0", default-features = false, features = ["render", "x11"] }
```
Comment (jomala) [inline Cargo.toml:18]: I'm not sure about these features - can you point me at a reference that indicates these are the right choices?

Are you deliberately making this "render" and the one above "bevy_render"?

**Review by jomala — COMMENTED** [review]
Thanks for the feedback. A few questions though ....

File: src/lib.rs (hunk 1 above)
Comment (SUPERCILEX) [reply src/lib.rs:93]: Shoot, you're right. Fixed by checking the update_now flag.

File: Cargo.toml (hunk 6 above)
Comment (SUPERCILEX) [reply Cargo.toml:18]: Yeah, it comes from this: https://github.com/bevyengine/bevy/blob/3e24b725afe143db5db006b146cd94339e155251/Cargo.toml#L36

`render` includes a bunch of extra crap that's needed to make things actually render, but those aren't needed to make things compile. I just added the minimal features needed to compile the lib.

File: src/lib.rs (hunk 5 above)
Comment (SUPERCILEX) [reply src/lib.rs:107]: Hmmm, I guess I hadn't thought about the accuracy thing. That's up to you, let me know what you prefer. I thought it'd be more important to show the FPS right away, but maybe not.

File: src/lib.rs (hunk 2 above)
Comment (SUPERCILEX) [reply src/lib.rs:126]: With `update_now`, the FPS shows one frame after the game starts. So depending on the decision on your previous comment, I'll add this back.

**Comment by SUPERCILEX** [comment]
@jomala Bump

**Comment by SUPERCILEX** [comment]
Another boop

**Comment by jomala** [comment]
Very sorry about going quiet, @SUPERCILEX . I've completely stopped all non-work coding over the last couple of months. I want to look at this properly, rather than just accepting it, but I'm just not getting to it. Hopefully in the next couple of weeks!

**Comment by SUPERCILEX** [comment]
No worries! Actually, this library should probably be deprecated now: https://github.com/bevyengine/bevy/pull/5084. :)

**Pull request closed without merging**